QGA-35-User-patch-route

Bekhzod96 opened the pull request:
DESCRIPTION:  USER COULD MODIFY ONLY USERNAME AND  PASSWORD

Bekhzod96 committed:
REFERANCE:QGA-35-User-patch-route
DESCRIPTION:  USER COULD MODIFY ONLY USERNAME AND  PASSWORD

Gruseckis commented on controllers/userController.js:
```diff
@@ -11,66 +11,81 @@ const getUserInfo = async (req, res) => {
+      }
+    }
+    if (req.user._id.toString() === id) {
+      const hashedPassword = await bcrypt.hash(
```
This will break code. Every time that user updates his information you will change password. This logic should be handled by pre hook

Bekhzod96 committed:
Merge branch 'development' of https://github.com/Gruseckis/quiz-game-api into QGA-35-User-patch-route

Bekhzod96 committed:
REFERANCE:QGA-35-User-patch-route
DESCRIPTION: I make changes when user send only username or password separetly and password will be saved hashed

Gruseckis commented on controllers/userController.js:
```diff
@@ -11,66 +11,91 @@ const getUserInfo = async (req, res) => {
-      throw new AppError('Only owner or admin can update the user');
     }
-  }catch(error){
+    if (req.user._id.toString() === id) {
```
Why tostring?

Bekhzod96 replied:
I haven't done it, it is Yana work

Gruseckis replied:
If you use some else functions you at least need to know what it does.

Gruseckis commented on controllers/userController.js:
```diff
@@ -11,66 +11,91 @@ const getUserInfo = async (req, res) => {
+      const keys = Object.keys(body);
+      let userUpdate = {};
+      let updatedUser;
+      keys.forEach(key => {
```
This should be done without forEach

Bekhzod96 committed:
REFERANCE:QGA-35-User-patch-route
DESCRIPTION: I wrote code withour forEach

Commit:
REFERENCE QGA-35
  DESCRIPTION Add functionality to control what user can chang about
  himself and user can chang only self.
